data-reconciliation-app/src/repositories/kv-repository.ts

```diff
@@ -0,0 +1,69 @@
+import * as ccfapp from "@microsoft/ccf-app";
+
+export interface IKeyValueRepository<T> {
+  set(key: string, value: T): T;
+  get(key: string): T;
+  has(key: string): boolean;
+  keys(): string[];
+  values(): T[];
+  get size(): number;
+}
+
+// generic key-value repository wrapping ccf TypedKvMap interaction
+export class KeyValueRepository<T> implements IKeyValueRepository<T> {
```

Comment:
Instead of exporting the class, we better export the singleton instance of it for better testability.

I'd organise this more like:

```typescript
export interface Repository<T> {
 ///////
}

class KeyValueRepository<T> implements Repository<T> {
     private readonly private kvStore: ccfapp.TypedKvMap<string, T>;

     constructor(kvStore: ccfapp.TypedKvMap<string, T>) {
         this.kvStore = kvStore;
     }
     ////////
}

const kvStore = ccfapp.typedKv("data", ccfapp.string, ccfapp.json<T>());
const keyValueRepository: Repository = new KeyValueRepository(kvStore);

export  keyValueRepository;
```

This above suggestion aims:

1. Exporting the interface as contract instead of concrete implementation (SOLID principales)
2. Helps spying on the exported singleton instance for unit testing
3. Injects the infrastructure dependency, `kvStore` at the constructor.

Reply:
Yeah, I agree; I already consume the interface. But exporting the concrete class to make it available for unit testing